frame/multisig/src/lib.rs

```diff
@@ -122,19 +92,81 @@ pub struct Multisig<BlockNumber, Balance, AccountId> {
 	approvals: Vec<AccountId>,
 }
 
-decl_storage! {
-	trait Store for Module<T: Config> as Multisig {
-		/// The set of open multisig operations.
-		pub Multisigs: double_map
-			hasher(twox_64_concat) T::AccountId, hasher(blake2_128_concat) [u8; 32]
-			=> Option<Multisig<T::BlockNumber, BalanceOf<T>, T::AccountId>>;
+type CallHash = [u8; 32];
 
-		pub Calls: map hasher(identity) [u8; 32] => Option<(OpaqueCall, T::AccountId, BalanceOf<T>)>;
-	}
+enum CallOrHash {
+	Call(OpaqueCall, bool),
+	Hash([u8; 32]),
 }
 
-decl_error! {
-	pub enum Error for Module<T: Config> {
+#[frame_support::pallet]
+pub mod pallet{
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// The overarching call type.
+		type Call: Parameter + Dispatchable<Origin=Self::Origin, PostInfo=PostDispatchInfo>
+			+ GetDispatchInfo + From<frame_system::Call<Self>>;
+
+		/// The currency mechanism.
+		type Currency: ReservableCurrency<Self::AccountId>;
+
+		/// The base amount of currency needed to reserve for creating a multisig execution or to store
+		/// a dispatch call for later.
+		///
+		/// This is held for an additional storage item whose value size is
+		/// `4 + sizeof((BlockNumber, Balance, AccountId))` bytes and whose key size is
+		/// `32 + sizeof(AccountId)` bytes.
+		#[pallet::constant]
+		type DepositBase: Get<BalanceOf<Self>>;
+
+		/// The amount of currency needed per unit threshold when creating a multisig execution.
+		///
+		/// This is held for adding 32 bytes more into a pre-existing storage value.
+		#[pallet::constant]
+		type DepositFactor: Get<BalanceOf<Self>>;
+
+		/// The maximum amount of signatories allowed in the multisig.
+		#[pallet::constant]
+		type MaxSignatories: Get<u16>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	/// The set of open multisig operations.
+	#[pallet::storage]
+	pub(super) type Multisigs<T: Config> = StorageDoubleMap<
```

Comment:
Should just be `pub` according to the old version:

```suggestion
	pub type Multisigs<T: Config> = StorageDoubleMap<
```

Reply:
The example pallet has the pub(super) I believe to export outside the module, it doesn't appear to matter here (not used outside module), so idc 

https://github.com/paritytech/substrate/blob/cf2d93115f72c70375157a89b75b046cdf491bd0/frame/example/src/lib.rs#L596-L600

Reply:
The nature of public interfaces is that you never really know who the dependents are. Suppose I created a new pallet that has the multisig pallet as a dependency, and I call into `Multisigs` in order to get/set data. If you were to restrict visibility to `pub(super)`, it would break this pallet.